components/carousel/render-media-item.tsx

```diff
@@ -0,0 +1,114 @@
+import { ActivityIndicator, Image, Platform } from "react-native";
+import Animated from "react-native-reanimated";
+import { ResizeMode, Video } from "expo-av";
+import { TapGestureHandler, State } from "react-native-gesture-handler";
+
+import { stylesFn } from "@/shared-uis/styles/InfluencerCard.styles";
+import { useTheme } from "@react-navigation/native";
+import { imageUrl } from "@/shared-uis/utils/url";
+import { useState } from "react";
+import { View } from "../theme/Themed";
+
+export interface MediaItem {
+  type: "image" | "video";
+  appleUrl?: string;
+  playUrl?: string;
```

Comment:
Can we please change these to only type and URL? We can decide at the time of passing props what we need to pass between appleUrl, playUrl, and imageUrl. But components should not have props like these.